### ISSUE TITLE
Potentially confusing description / changelog about field method names

### ISSUE THREAD

**Issue by grantbdev** [rmosolgo/graphql-ruby#3054]
In [v1.11.0](https://github.com/rmosolgo/graphql-ruby/blob/master/CHANGELOG.md#1110-13-june-2020), fields defined in camel case (`field :doStuff`) no longer lookup methods that are underscore case (`def do_stuff`). The changelog suggests that one way of fixing this is by adding `method: :do_stuff` as an option to the field to explicitly map the field to a method.

These options were already in use in my codebase and I assumed they were working, but in 1.11.0 I would get an error like the following:
```
Failed to implement Query.doStuff, tried:

- `Types::QueryType#doStuff`, which did not exist
- `NilClass#do_stuff`, which did not exist
- Looking up hash key `:all_companies` or `"all_companies"` on ``, but it wasn't a Hash

To implement this field, define one of the methods above (and check for typos)
```

Upon further digging in the [documentation](https://graphql-ruby.org/fields/introduction.html#field-resolution), I discovered there is an alternative field option `resolver_method` which did work for me in this case. I was confused about the distinction, but I realized the difference is whether the named method is located in the current class or in the underlying object being represented, if there is one. For the query type, I guess there is not an underlying object hence the attempt at `NilClass#do_stuff` and so `resolver_method` must be used instead of `method`. I understand this now, but even looking at the documentation I was confused for a little bit since by default the field name can automatically map to a method in the current class or in the underlying object.

It would be ideal if `method` and `resolver_method` could be converged, but if that's not technically possible it might be worth mentioning in the changelog for the 1.11.0 release since `method` is not always the right option. At least, I hope this issue is useful to the next person who runs into this problem.

**Comment by swalkinshaw** [comment]
For a bit of history, they used to be converged before and `method` handled both 😅 I disambiguated them in  https://github.com/rmosolgo/graphql-ruby/pull/1961

**Comment by rmosolgo** [comment]
Thanks for pointing out this issue! I think you're spot on that it should have been `resolver_method`.  As @swalkinshaw  points out, there's good reason to have them separate ... but apparently I can't keep them straight 🙈 . I fixed the suggestion in c81aa65, if you think it should be improved further, please open a PR!
